core/src/main/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContext.java

```diff
@@ -112,7 +112,9 @@ private void addArgumentsAsVariables() {
 		}
 
 		for (int i = 0; i < args.length; i++) {
-			super.setVariable(paramNames[i], args[i]);
+			if (paramNames[i] != null) {
+				super.setVariable(paramNames[i], args[i]);
```

Comment:
For the test below to work, you need to remove `super.`

Reply:
Removed super.